platforms/ide/problems/src/main/java/org/gradle/problems/internal/transformers/PluginIdLocationTransformer.java

```diff
@@ -36,20 +37,21 @@ public PluginIdLocationTransformer(
 
     @Override
     public Problem transform(Problem problem) {
-        getExecuteTask(OperationDetails.class)
-            .ifPresent(id -> {
+        return getExecuteTask(OperationDetails.class)
+            .map(id -> {
                 try {
                     OperationDetails operationDetails = operationListener.getOp(id, OperationDetails.class);
                     Objects.requireNonNull(operationDetails, "operationDetails should not be null");
                     String pluginId = operationDetails.getPluginId();
                     if (pluginId != null) {
-                        problem.getLocations().add(new PluginIdLocation(pluginId));
+                        ProblemCloneBuilder cloneBuilder = problem.createCloneBuilder();
+                        cloneBuilder.getLocations().add(new PluginIdLocation(pluginId));
```

Comment:
❌ 
This is weird. Can we replace it with `cloneBuilder.pluginLocation(pluginId)`? cloneBuilder implements `ProblemBuilderDefiningLocation`, right? If the method does not exist there, I think it should.